posts/2021-03-25-Rust-1.51.0.md

```diff
@@ -0,0 +1,179 @@
+
+This has been a long standing issue in cargo, and with this release there's a new `resolver` option in your `Cargo.toml`, where you can set `resolver="2"` to tell cargo to try a new approach to resolving features, you can check out [RFC 2957] a detailed description of the behaviour, but it can summarised as follows.
+
+- **Dev dependencies** — When a package is shared as a normal dependency and a dev-dependency, the dev-dependency features are only enabled if the current build is including dev-dependencies.
+- **Host Dependencies** — When a package is shared as a normal dependency and a build-dependency or proc-macro, the features for the normal dependency are kept independent of the build-dependency or proc-macro.
+- **Target dependencies** — When a package appears multiple times in the build graph, and one of those instances is a target-specific dependency, then the features of the target-specific dependency are only enabled if the target is currently being built.
+
+While this can lead to a compiling some crates more than once, this should provide a much more intuitive development experience when using features with cargo. If you'd like to know more, you can also read the ["Feature Resolver"][feature-resolver@2.0] section in the Cargo Book for more information. We'd like to thank the cargo team and everyone involved for all their hard work in designing and implementing this new feature (resolver)!
+
+[feature-resolver@2.0]: https://doc.rust-lang.org/nightly/cargo/reference/features.html#feature-resolver-version-2
+
+### Splitting Debug Information
+While not often highlighted in the release, the Rust teams are constantly working on improving compile Rust's compile times, and this release marks one of the biggest improvements in a long time for Rust on macOS platforms. Debug information is code that maps the binary code back to your source code, so that the program can give you more information about what when wrong at run time. In macOS this is done with a tool called `dsymutil` which collects all the debug information into a single `.dSYM` file.
+
+This tool provides a lot of benefits, however one of the drawbacks has been that `dsymutil` is not really compatible with incremental recompilation, so even when you make a small change such as a function name, `dsymutil` will need to run over the entire final binary to produce the final `.dSYM` folder. This can add a lot to the build time, especially for larger projects, however this has been a necessary step as without it Rust wouldn't be able to parse debug info on macOS.
+
+However with the implementation of the Rust based `backtrace` crate backend, we're now able to parse debuginfo even without running `dsymutil`. This can significantly speed up builds that include debuginfo. We haven't ran extensive benchmarks but we've seen a lot of anecdata of people's builds building a lot faster on macOS with this behaviour.
```

Comment:
nit: you say 'however' quite a lot, maybe change the wording a little?
```suggestion
This tool provides a lot of benefits, however one of the drawbacks has been that `dsymutil` is not really compatible with incremental recompilation. That means even when you make a small change such as a function name, `dsymutil` will need to run over the entire final binary to produce the final `.dSYM` folder. This can add a lot to the build time, especially for larger projects, but this has been a necessary step as without it Rust wouldn't be able to parse debug info on macOS.

With the implementation of the Rust based `backtrace` crate backend, we're now able to parse debuginfo even without running `dsymutil`. This can significantly speed up builds that include debuginfo. We haven't run extensive benchmarks but we've seen a lot of anecdata of people's builds building a lot faster on macOS with this behaviour.
```